public/js/sidebar.js

```diff
@@ -231,11 +276,47 @@ export default class extends Component {
             </div>
         );
     }
+
+    sharingPanelJSX() {
+        return (
+            <div className="sharing-panel">
+                <div className="section-header-sidebar">
+                    <h4>
+                        Share results
+                    </h4>
+                </div>
+                <ul className="nav">
+                    {
+                        <li>
+                            <a id="copyTooltip" className ="btn-link copy-URL" data-toggle="tooltip"
+                                onClick={this.copyURL}> 
+                                <i className="fa fa-copy"></i> Copy URL to clipboard
+                            </a> 
+                        </li>  
+                    } 
+                    { 
+                        <li>
+                            <a id="sendEmail" className ="btn-link email-URL" data-toggle="tooltip"
+                                title="Send by email" href= {`mailto:?subject=SeqServ results &body=Thank you for using SequenceServer.
+                                %0DBelow, you will find a link to the results of your most recent search. While using SequenceServer, you may use this link to access previous results.
+                                %0DYou will also find the unique ID of your query and the first 15 databases used in your search.
+                                %0D%0DLink: `+ window.location.href + '%0DQuery id: ' + this.props.data.search_id +
+                                '%0DDatabases:'  + this.usedDatabases() + '%0D%0DPlease cite: https://doi.org/10.1093/molbev/msz185'}
+                                target="_blank" rel="noopener noreferrer">
+                                <i className="fa fa-envelope"></i> Send by email
+                            </a>
```

Comment:
I think you are better off extracting href's value into a function of its own (maybe call it `mailtoLink`. Use multiline string in the function and `urlQuoteParams` or other suitable function to automatically turn special characters to `%0D` etc.

Reply:
Initially, ` href ` was in a function that was assigned to onClick in this `a` element. It was changed to the current format to access the `href` for testing (Capybara_rspec.rb, 222). I can revert as suggested but could you advise on how to check the `href` contents in the function for testing?

Reply:
Not sure I fully understand your concern. You would access href in tests like you do now. You don’t need to assign href on click. It’s a fixed string for a given results page. You just construct it in a function and assign the return value to href.

    mailtoLink () {
      return ‘mailto:blah’
    }

    <a href={this.mailtoLink()}>Share</a>

Reply:
Did you inspect the output of `encodeURI` to see if there is `%09` in it?

Maybe this is what you wanted: `message.replace(/(%20){2,}/g, '')`.

But maybe npm packages like dedent are also a good alternative.